Documentation/Storage-Configuration/Object-Storage-RGW/cosi.md

```diff
@@ -200,13 +199,6 @@ spec:
     volumeMounts:
     - name: cosi-secrets
       mountPath: /data/cosi
-  initContainers:
```

Comment:
I removed the init container as we don't have `setup-aws-credentials` in busybox container. any alternative image for this one? if not can were remove the annotation approach mentioned in the doc?

Reply:
Any bash scripting container image will do, I though busybox is light weight container. Idea here is provide a way  to parse the BucketInfo as Enviroment variables

Reply:
@thotz Thanks, am not sure any image will have the `setup-aws-credentials` CLI command in it. if you can specify the exact container image which contains `setup-aws-credentials` will replace it, for now this pod is failing to start with below error

```
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason     Age              From               Message
  ----     ------     ----             ----               -------
  Normal   Scheduled  7s               default-scheduler  Successfully assigned default/sample-app to minikube
  Normal   Pulling    5s (x2 over 6s)  kubelet            Pulling image "busybox"
  Normal   Pulled     5s               kubelet            Successfully pulled image "busybox" in 1.117956858s (1.117975281s including waiting)
  Normal   Created    4s (x2 over 5s)  kubelet            Created container init-cosi
  Normal   Started    4s (x2 over 5s)  kubelet            Started container init-cosi
  Normal   Pulled     4s               kubelet            Successfully pulled image "busybox" in 1.074652516s (1.074667163s including waiting)
  Warning  BackOff    2s (x2 over 3s)  kubelet            Back-off restarting failed container init-cosi in pod sample-app_default(98bcef58-61a8-4a14-9558-4741780afc9b)
[🎩︎]mrajanna@li-2cfbef4c-22d9-11b2-a85c-a3e4a93c405f cosi $]kubectl logsectl logs po/sample-app
Defaulted container "sample-app" out of: sample-app, init-cosi (init)
Error from server (BadRequest): container "sample-app" in pod "sample-app" is waiting to start: PodInitializing
[🎩︎]mrajanna@li-2cfbef4c-22d9-11b2-a85c-a3e4a93c405f cosi $]kubectl logs po/sample-app init-cosi
/bin/sh: setup-aws-credentials: not found
[🎩︎]mrajanna@li-2cfbef4c-22d9-11b2-a85c-a3e4a93c405f cosi $]
```

Reply:
Can we mount the secret as environment variables directly? Or is the COSI secret not compatible with that option?

Reply:
not sure about that @thotz any idea about it?

Reply:
It is .json format. We need to parse cannot export directly as env like OBC

Reply:
Up on lines 177-180 we have the example to export the env vars, but if that is in an init container, the variables would only apply to the init container and wouldn't be passed on to the main container. So it seems the app will just need to parse the json.